cmd/manager/main.go

```diff
@@ -311,3 +315,15 @@ type finalizerFunc func(ctx context.Context, obj client.Object) (crfinalizer.Res
 func (f finalizerFunc) Finalize(ctx context.Context, obj client.Object) (crfinalizer.Result, error) {
 	return f(ctx, obj)
 }
+
+func authFilePathIfPresent(logger logr.Logger) string {
+	_, err := os.Stat(authFilePath)
+	if os.IsNotExist(err) {
+		return ""
+	}
+	if err != nil {
+		logger.Error(err, "could not stat auth file path", "path", authFilePath)
+		return ""
```

Comment:
Are we okay with not panicking here if an unknown stat error occurs here?

Reply:
Yeah, you're probably right. Should we have this function return `(string, error)`?  That makes calling it slightly more verbose to call, but then we could do the error check and have the logging/os.Exit happen in main. And at that point, no need to pass a logger in.

Reply:
I like the idea of keeping the signature the same and then `os.Exit(1)` instead of `return ""`. Updated the PR.